pkg/controller/servicebindingrequest/mapper.go

```diff
@@ -12,26 +21,170 @@ var (
+
+// add adds the given namespaced name n into the set.
+func (t namespacedNameSet) add(n types.NamespacedName) {
+	t[n] = true
+}
+
+func convertToRequests(t namespacedNameSet) []reconcile.Request {
+	toReconcile := make([]reconcile.Request, 0)
+	for n := range t {
+		toReconcile = append(
+			toReconcile,
+			reconcile.Request{NamespacedName: n},
+		)
+	}
+	return toReconcile
+}
 
 // Map execute the mapping of a resource with the requests it would produce. Here we inspect the
 // given object trying to identify if this object is part of a SBR, or a actual SBR resource.
+//
+// This method is responsible for ingesting arbitrary Kubernetes resources (for example corev1.Secret
+// or appsv1.Deployment) and lookup whether they are related to one or more existing Service Binding
+// Request resources.
 func (m *sbrRequestMapper) Map(obj handler.MapObject) []reconcile.Request {
 	log := mapperLog.WithValues(
 		"Object.Namespace", obj.Meta.GetNamespace(),
 		"Object.Name", obj.Meta.GetName(),
 	)
-	toReconcile := []reconcile.Request{}
 
-	sbrNamespacedName, err := getSBRNamespacedNameFromObject(obj.Object)
+	namespacedNamesToReconcile := make(namespacedNameSet)
+
+	if isServiceBindingRequest(obj.Object) {
+		requests := []reconcile.Request{
+			{NamespacedName: convertToNamespacedName(obj.Meta)},
+		}
+		log.Debug("current resource is a SBR", "Requests", requests)
+		return requests
+	}
+
+	// note(isutton): The client handles retries on the operator behalf, so only unrecoverable errors
+	// are left.
+	//
+	// please see https://github.com/isutton/service-binding-operator/blob/e17445570bd3889bcf7499142350a3b81463c6be/vendor/k8s.io/client-go/rest/request.go#L723-L812
+	sbrList, err := m.client.Resource(groupVersion).List(metav1.ListOptions{})
 	if err != nil {
-		log.Error(err, "on inspecting object for annotations for SBR object")
-		return toReconcile
+		log.Error(err, "listing SBRs")
+		return []reconcile.Request{}
 	}
-	if isNamespacedNameEmpty(sbrNamespacedName) {
-		log.Debug("not able to extract SBR namespaced-name")
-		return toReconcile
+
+ITEMS:
+	for _, item := range sbrList.Items {
+		namespacedName := convertToNamespacedName(&item)
+
+		sbr, err := convertToSBR(item.Object)
+		if err != nil {
+			log.Error(err, "converting unstructured to SBR")
+			continue ITEMS
+		}
+
+		if isSecret(obj.Object) && isSecretOwnedBySBR(obj.Meta, sbr) {
+			log.Debug("resource identified as a secret owned by the SBR")
+			namespacedNamesToReconcile.add(namespacedName)
+		} else {
+			log.Debug("resource is not a secret owned by the SBR")
+		}
+
+		if isSBRService(sbr, obj.Object) {
+			log.Debug("resource identified as service in SBR")
+			namespacedNamesToReconcile.add(namespacedName)
+		} else {
+			log.Debug("resource is not a service declared by the SBR")
+		}
+
+		if ok, err := isApplication(m.restMapper, sbr, obj.Object); err != nil {
+			log.Error(err, "identifying resource resource as SBR application")
```

Comment:
Typo -> identifying resource resource

Reply:
Solved in ba54c6237f5d22cbaba2bb10e293aae6d518b253.